Add some clarification on generated source code being derived artifacts (#35)
omnibor/site: content/glossary/build_tool.md

    compiler --> .pyc
```

The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).

## Code Generators

It is a not infrequent occurence to have build tools which generate source code from an input.

```mermaid
flowchart LR
    codegeninput[codegen input] --> codegenerator[[code generator]] --> generatedsrc[generated source code file]
```

> The source code files output by a code generator are themselves [derived artifacts](http://localhost:1313/glossary/artifact/#derived-artifacts) and as such are no longer [leaf artifacts](http://localhost:1313/glossary/artifact/#leaf-artifacts) in the artifact tree

Reviewer: > and as such are no longer leaf artifacts in the artifact tree

Could you expand / clarify this point?

Author: I think the rewrite may bring more clarity :)

    compiler --> .pyc
```

The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).

## Code Generators

Typically, source code files are hand written by humans, and as such are [leaf artifacts](http://localhost:1313/glossary/artifact/#leaf-artifacts) in the [artifact tree](/glossary/artifact_tree).

> Sometimes source code files are themselves **generated** by a code generator from inputs.

Reviewer 2: ```suggestion
Source code files can also be **generated** from other inputs by a code generator.
```

Author: Done.

```mermaid
flowchart LR

>     codegeninput[codegen input] --> codegenerator[[code generator]] --> generatedsrc[generated source code file]

Reviewer 2: Took a minute to parse the semantics behind this. It's probably more important to simply highlight that this is an 'input', rather than to label it a 'codegen input'. Suggest deleting 'codegen'.

Author: Done

```

> 

Reviewer 2: It would be good to have an example, or a link to an example to help solidify the concept.

Author: Done.

> A code generator is a [build tool](#build-tools).  The output from a [build tool](#build-tools) is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).  Therefore generated soure code is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).

Reviewer 2: Suggest rewriting this to get to the point first, then explain why. E.g. 
```suggestion
In this scenario, generated source code is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts). This is because the [code generator](build_tool/#code-generators) is a [build tool](#build-tools) and, by definition, the output from the [build tool](#build-tools) is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).
```

Author: Done.
